### PR TITLE
[DRFT-262] Fix bulk select in add system modal

### PR DESCRIPTION
**Pull request by johnsonm325**
To test:

- Navigate to Add to comparison modal
- Click Bulk Selector to select all
- click the Bulk Selector again

The full set of systems on page is added every time we click, making the basket contain duplicates. If we then remove a system from basket, also it's duplicate is removed.

This PR makes sure that when a system, baseline or historical profile is selected, but not compared, and you back out of the add system modal screen, those items are removed from the basket and all selections in the table, when you return to the add system modal, are removed.

This PR also ensures that the bulk selector properly adds and removes systems to the selected basket in the top right corner of the add system modal.

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by johnsonm325** [comment]
@vkrizan tests are failing, I need to write new tests, and there are conflicts; but you should be able to test this PR with no problem.

File: src/SmartComponents/AddSystemModal/AddSystemModal.js [hunk 1]
```diff
@@ -153,19 +171,23 @@ export class AddSystemModal extends Component {
...
-                if (row.id === data.id) {
-                    selectedSystems.push({ id: row.id, name: row.display_name, icon: <ServerIcon /> });
-                }
-            });
+            if (!data.selected) {
+                selectedSystems = selectedSystemContent.filter(system => system.id === data.id);
+            } else {
+                entities.rows.forEach(function(row) {
+                    if (row.id === data.id) {
+                        selectedSystems.push({ id: row.id, name: row.display_name, icon: <ServerIcon /> });
```
Comment (vkrizan) [inline src/SmartComponents/AddSystemModal/AddSystemModal.js:187]: Observation: can combine `filter` and `map`, to have it w/o `push`.

File: src/SmartComponents/AddSystemModal/AddSystemModal.js [hunk 2]
```diff
@@ -123,8 +123,26 @@ export class AddSystemModal extends Component {
...
+        }
+    }
+
     cancelSelection() {
-        const { toggleAddSystemModal } = this.props;
+        const { baselines, handleBaselineSelection, handleHSPSelection, handleSystemSelection, historicalProfiles,
+            selectBaseline, selectedBaselineContent, selectedHSPContent, selectedSystemContent, selectHistoricProfiles,
+            systems, toggleAddSystemModal } = this.props;
+
+        handleSystemSelection(this.findNotInComparison(selectedSystemContent, systems), false);
```
Comment (vkrizan) [inline src/SmartComponents/AddSystemModal/AddSystemModal.js:139]: Observation: The system selection (number) on the bulk select is visible for a brief moment when reopening the modal. Then is reset.

File: src/SmartComponents/SystemsTable/SystemsTable.js [hunk 3]
```diff
@@ -115,7 +114,7 @@ export const SystemsTable = ({
                     } }
                 bulkSelect={ onSelect && !isAddSystemNotifications && {
                     isDisabled: !hasMultiSelect,
-                    count: entities && entities.selectedSystemIds ? entities.selectedSystemIds.length : 0,
+                    count: entities?.selectedSystems?.length,
```
Comment (vkrizan) [inline src/SmartComponents/SystemsTable/SystemsTable.js:117]: This shows only selected items on the viewed page and not the whole selection.

Screenshot_2021-06-08 Comparison - Drift Red Hat Insights
(here the bulk select is completely empty, check the selection on top right)

Comment (vkrizan) [reply src/SmartComponents/SystemsTable/SystemsTable.js:117]: After I got an error (selecting 51 systems) for comparison and opening the modal again, I see the selection is visible in the basked however it is not displayed on the table nor on the bulk select.

Screenshot_2021-06-08 Comparison - Drift Red Hat Insights(1)
(I had first page selected and then one item on next page)

Comment (johnsonm325) [reply src/SmartComponents/SystemsTable/SystemsTable.js:117]: > This shows only selected items on the viewed page and not the whole selection.
> 
> Screenshot_2021-06-08 Comparison - Drift Red Hat Insights
> (here the bulk select is completely empty, check the selection on top right)

If I understand it correctly, this is what we expect. If you make a full selection on page 1, and then go to page 2, if nothing on page 2 is selected you should see the number of selected from page 1 in the selection in the top right, but the bulk select should be unchecked.

Comment (vkrizan) [reply src/SmartComponents/SystemsTable/SystemsTable.js:117]: We implemented the bulk select recently, and the behavior we used was the one I've described (whole selection). What is the source of the expectation for the bulk to display current page selection? It feels a bit strange TBH.
Maybe @bastilian can chime in?

Comment (johnsonm325) [reply src/SmartComponents/SystemsTable/SystemsTable.js:117]: none selected = empty check
some selected = dash
all selected = check

I did review with UX, and they were okay with the use case.

Comment (bastilian) [reply src/SmartComponents/SystemsTable/SystemsTable.js:117]: Not sure what I should chime in about or how. :)

The bulk select with the selection of all items indicated outside and only having page selection operations, is different from how we implemented it on our rules table, but if UX approves of it it should be ok.

Comment (vkrizan) [reply src/SmartComponents/SystemsTable/SystemsTable.js:117]: @johnsonm325 If UX agrees, then it is all good. 
Can you please check the issue described in the second comment https://github.com/RedHatInsights/drift-frontend/pull/481#discussion_r647232493?

**Comment by johnsonm325** [comment]
@vkrizan I ran the bulk select by UX again and it was a misunderstanding. Because our system table pulls all systems available, we have to treat the bulk select a bit differently. I'll try to outline:

- Clicking box on page 1 selects all systems on that page and replaces box with dash
- Selecting it again deselects the page and emptys the box if nothing else is selected
- Selecting all of page 1 and partial page 2 will show total collected across all pages and leave the dash
- Selecting all on every page will replace with a check
- Selecting `Select none` from the bulk select dropdown removes all selections across all pages
- Selecting `Select page` from the bulk select dropdown selects all on the current page
- Selecting `Deselect page` from the bulk select dropdown deselects all on the current page

I'm going to write tests in another PR. I want to get this feature with fixes to prod before vacation next week. Hopefully everything works. Thanks again!

**Review by vkrizan — APPROVED** [review]
Awesome! Works great.

Thank you Michael.

**Pull request merged**